File: almostIncreasingSequence.js
function almostIncreasingSequence(sequence) {
    var needRemoval = 0;
    for (var i = 0; i < sequence.length; i++) {
        if (sequence[i] <= sequence[i - 1]) {
            needRemoval++;
        }
        if (needRemoval > 1) {
            return false;
        }
        if (sequence[i] <= sequence[i - 2] && sequence[i + 1] <= sequence[i - 1]) {
            return false;
        }
    }
    return true;
}

File: 1_intro/2_edge_of_the_ocean/7_almostIncreasingSequence/almostIncreasingSequence.lua
function almostIncreasingSequence(sequence)
  needRemoval = 0
  len = table.getn(sequence)
  for i = 2, len do
    if sequence[i] <= sequence[i - 1] then
      needRemoval = needRemoval + 1
    end
    if needRemoval > 1 then
      return false
    end
    if (i >= 3) and (i < len) then
      if (sequence[i] <= sequence[i - 2]) and (sequence[i + 1] <= sequence[i - 1]) then
        return false
      end
    end
  end
  return true
end